app/src/main/java/com/rai/hearthstonecard/adapter/CardAdapter.kt

```diff
@@ -8,61 +8,38 @@ import androidx.recyclerview.widget.ListAdapter
 import androidx.recyclerview.widget.RecyclerView
 import coil.load
 import com.rai.hearthstonecard.databinding.ItemCardBinding
-import com.rai.hearthstonecard.databinding.ItemLoadingBinding
-import com.rai.hearthstonecard.model.Item
+import com.rai.hearthstonecard.domain.model.Card
 
 
 class CardAdapter(
     context: Context,
-    private val onItemClicked: (Item.Content) -> Unit,
-) : ListAdapter<Item, RecyclerView.ViewHolder>(DIFF_UTIL) {
+    private val onItemClicked: (Card) -> Unit,
+) : ListAdapter<Card, CardViewHolder>(DIFF_UTIL) {
 
-    //есть устойчивое чувство что что то не то делаю с Item, но вроде работает
     private val layoutInflater = LayoutInflater.from(context)
 
-    override fun getItemViewType(position: Int): Int {
-        return when (getItem(position)) {
-            is Item.Content -> CARD_TYPE
-            Item.Loading -> LOADING_TYPE
-            else -> error("Incorrect item")
-        }
-    }
 
-    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
-        return when (viewType) {
-            CARD_TYPE -> {
-                CardViewHolder(
-                    binding = ItemCardBinding.inflate(layoutInflater, parent, false)
-                )
-            }
-            LOADING_TYPE -> {
-                LoadingViewHolder(
-                    binding = ItemLoadingBinding.inflate(layoutInflater, parent, false)
-                )
-            }
-            else -> error("Incorrect view type $viewType")
-        }
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CardViewHolder {
+        return CardViewHolder(
+            binding = ItemCardBinding.inflate(layoutInflater, parent, false)
+        )
+
     }
 
-    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
-        val cardViewHolder = holder as? CardViewHolder ?: return
-        val item = getItem(position) as? Item.Content ?: return
-        cardViewHolder.bind(item) {
+    override fun onBindViewHolder(holder: CardViewHolder, position: Int) {
+        val item = getItem(position)
+        holder.bind(item) {
```

Comment:
у тебя в метод bind передается лямбда, можно проще 
```suggestion
        holder.bind(item, onItemClicked)
```